### ISSUE TITLE
Cannot install prefect<1.0.0

### ISSUE THREAD

**Issue by jtanx** [astral-sh/uv#1798]
uv version: 0.1.6
uv platform: almalinux 8/windows

```
>uv pip install "prefect<1.0.0"
error: Failed to download: prefect==0.15.13
  Caused by: Couldn't parse metadata of prefect-0.15.13-py3-none-any.whl from https://files.pythonhosted.org/packages/42/1f/ef0a4440c76848820b468f78d31f02d8b75e9bceea398df8fcef1d775713/prefect-0.15.13-py3-none-any.whl
  Caused by: after parsing 9.0.0a1, found ".0" after it, which is not part of a valid version
kubernetes (>=9.0.0a1.0) ; extra == 'all_extras'
            ^^^^^^^^^^^
```

**Comment by konstin** [comment]
The version is invalid, but we should probably skip those packages instead of erroring (#1800).

**Comment by jtanx** [comment]
pip seems to parse that version specifier as 9.0.0a1 which does correspond to an actual version; is it just truncating the invalid part?

**Comment by konstin** [comment]
Yes, pip is currently ignoring everything in a version they can't parse ("legacy version"), but they also intend to switching to ignoring those files, too.

**Comment by zanieb** [comment]
Oops, that's well within my tenure :)

**Comment by charliermarsh** [comment]
Should be fixed in v0.1.7 (out now).

**Comment by jtanx** [comment]
Nice thanks, looks good!
